fs/vfs/fs_epoll.c

```diff
@@ -730,22 +823,28 @@ int epoll_wait(int epfd, FAR struct epoll_event *evs,
 #endif
 
       ret = nxsem_tickwait(&eph->sem, ticks);
-      if (ret == -ETIMEDOUT)
-        {
-          ret = OK;
-        }
     }
   else
     {
       ret = nxsem_wait(&eph->sem);
     }
 
-  if (ret < 0)
+  if (ret < 0 && ret != -ETIMEDOUT)
     {
       goto err;
     }
+  else /* ret >= 0 or ret == -ETIMEDOUT */
+    {
+      int num = epoll_teardown(eph, evs, maxevents);
+      if (num == 0 && timeout != 0 && ret != -ETIMEDOUT)
```

Comment:
```suggestion
      if (num == 0 && ret != -ETIMEDOUT)
```

Reply:
When `timeout == 0`, there is an endless loop from `retry:` to `goto retry;`

Reply:
Ok

Reply:
but let's change line 810 to `ret = -ETIMEOUT;`:
```suggestion
      if (num == 0 && ret >= 0)
```

Reply:
Done